File: 백준/Silver/14248. 점프 점프/점프 점프.py
import sys
data_temp = sys.stdin if sys.platform == 'linux' else open('입력.txt', 'r')
input_data = data_temp.read().splitlines()
from collections import deque

def solution (data) :
  [n], arr, [start] = [list(map(int, x.split())) for x in data]
  visited = [0] * n
  
  queue = deque([start - 1])
  
  while queue : 
    node = queue.popleft()
    visited[node] = True
    
    nextNode = [node + arr[node], node - arr[node]]
    
    for v in nextNode :
      if 0 <= v < n and not visited[v] :
        queue.append(v)

  print(visited.count(1))
    
solution(input_data)

File: 백준/Silver/11060. 점프 점프/점프 점프.js
const fs = require("fs");
const input = fs
  .readFileSync(process.platform === "linux" ? "/dev/stdin" : "입력.txt")
  .toString()
  .trim()
  .split("\n");

function solution(data) {
  const N = +data[0];
  const arr = data[1].split(" ").map(Number);

  const visited = Array.from({ length: N }, () => 0);
  const queue = [[0, 1]];
  let queueIdx = 0;
  let result = -1;

  while (queueIdx < queue.length) {
    const [idx, cnt] = queue[queueIdx++];
    if (idx === N - 1) {
      result = cnt - 1;
      break;
    }
    const node = arr[idx];
    if (visited[idx]) continue;
    visited[idx] = cnt;

    for (let i = idx; i <= idx + node; i++) {
      if (i > N - 1) continue;
      if (!visited[i]) queue.push([i, cnt + 1]);
    }
  }

  console.log(result);
}

solution(input);
